sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/CnameRecordSetImpl.java

```diff
@@ -27,6 +27,13 @@ public String canonicalName() {
 
     @Override
     protected RecordSetInner prepareForUpdate(RecordSetInner resource) {
+        if (resource.cnameRecord() == null) {
+            resource.withCnameRecord(new CnameRecord());
+        }
+        if (innerModel().cnameRecord().cname() != null) {
+            resource.cnameRecord().withCname(innerModel().cnameRecord().cname());
+        }
+        innerModel().withCnameRecord(new CnameRecord());
```

Comment:
Should the code follow the same pattern as other types (especially the removal part as `recordSetRemoveInfo`)?

E.g. ARecord
https://github.com/Azure/azure-sdk-for-java/blob/b17e959f998caee8a5e86b7340276b4d98078db1/sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/ARecordSetImpl.java#L37-L57

Reply:
OK I see. There is only 1 cnameRecord.

Reply:
Yes. I have modelled it after SoaRecordSetImpl, trying not to introduce any new patterns etc.